### PR TITLE
General Support for Multiple Classes and Relative Gap using CLI

### PR DESCRIPTION
**Pull request by ribsthakkar**
Support for multiple demand files beyond just NCTCOG.
Also can pass in relative gap using CLI

### CODE REVIEW

**Comment by sboyles** [comment]
When main.c was updated for the 2 new arguments, we changed all the old argv[] indices except the one in line 150 (old line 149)... when setBatches is called it still refers to argv[2].  Is that intentional?

**Comment by ribsthakkar** [comment]
> When main.c was updated for the 2 new arguments, we changed all the old argv[] indices except the one in line 150 (old line 149)... when setBatches is called it still refers to argv[2]. Is that intentional?

that was not intentional. I am pushing a fix addressing it in a different way since that argv[]==null implies we warm start if no demand file was provided. Howver, if no second argument was provided it also means that the program will segfault when accessing that memory. In this case, since we accept an arbitrary number of demand files. I explicitly search for 0. On the other hand, we should have been searching for "" or something that represents an empty string to avoid segmentation faults.

File: src/fileio.c
Comment (pyrito) [inline src/fileio.c:829]: Do we need this?

File: src/main.c
Comment (pyrito) [inline src/main.c]: While we can make gap configurable, I think it'd make more sense to keep a default value and then let someone change it if they really wanted to. (If someone feels strongly about this let me know)

**Review by pyrito — COMMENTED** [review]
I have an idea to change the argument parsing in general because using argv and argc is a real pain in the butt. My internship ends tomorrow so I'll probably start working on this on Saturday. We can hold off until then.

File: src/fileio.c
Comment (ribsthakkar) [reply src/fileio.c:829]: This is responsible for printing the flows of each individual class, part of PD’s project. But I still think it makes sense because we print the total flow and then broken down by class.

**Comment by ribsthakkar** [comment]
This is supposed to be a temporary fix. Long term we need a config file that sets of our args and configuration.

File: src/fileio.c
Comment (pyrito) [reply src/fileio.c:829]: I agree, can we make this a debug output instead? For example, we could have someone toggle `--verbose` and it can toggle to debug mode or something along those lines.

**Comment by pyrito** [comment]
> This is supposed to be a temporary fix. Long term we need a config file that sets of our args and configuration.

Ok I agree with that train of thought. I think let's try better arg parsing and then move towards integrating the configuration model, which I think would be useful more for the NCTCOG things (file names and such). Let's go ahead and merge this in the meantime if there are no edge cases that can break this. We need to update the README.md to reflect these changes as well.

**Review by pyrito — APPROVED** [review]

**Pull request merged**
